### DISCUSSION TITLE
"Some files have not passed the integrity check"

### DISCUSSION THREAD

**Discussion by iGadget** [nextcloud/all-in-one#2711] [Questions] [answered]
Not sure since when my AIO instance is experiencing this issue (you only see it when you access the Administration -> Overview page), but this page now tells me "Some files have not passed the integrity check".

Here are the details:
```
Results
=======
- core
	- INVALID_HASH
		- core/js/mimetypelist.js
	- EXTRA_FILE
		- core/img/filetypes/dwb.svg
		- core/img/filetypes/drawio.svg

Raw output
==========
Array
(
    [core] => Array
        (
            [INVALID_HASH] => Array
                (
                    [core/js/mimetypelist.js] => Array
                        (
                            [expected] => 4b0a99fbab7bc4824176101150bd0ab2b553e128a390e0f872580db4541462b31aad13099fd41533b6695d1b185dabca26a45e9a78227cfb9479eafa439c7bff
                            [current] => 0db33a45988de38e826762b7a452f88bf3aec65838ebc7e89ae9b6ecf82d39bae15efd9018c7f83bc04517030efcd6fb565070173d0b55fa3066e1b200233fe1
                        )

                )

            [EXTRA_FILE] => Array
                (
                    [core/img/filetypes/dwb.svg] => Array
                        (
                            [expected] => 
                            [current] => 43731dd5f17a048112ea5109b40b02ec019b3ee2324385a0f448e3bd2264cb13dc160ab018d893f92f8e2f168fd09009b51578c8c6b97a02a1617c67ac087701
                        )

                    [core/img/filetypes/drawio.svg] => Array
                        (
                            [expected] => 
                            [current] => 92e0974cf869bf8ab969c3442dc2b80d55fde36441d22924db74916a06b407520aa2a9dc39336f9157195ebede697ffac0e639360879255ab91932d406e1897d
                        )

                )

        )

)
```
Anyone else experiencing this? And how should I fix this? The [docs](https://docs.nextcloud.com/server/26/admin_manual/issues/code_signing.html) state that I should "upload the correct files"... But how should I go about that?

**Comment by szaimen** [comment, marked as answer]
Hi, I am pretty sure that this is caused by the drawio app. This should bw fixed automatically after disabling the app and waiting for the next Nextcloud update getting installed. For further help on this please refer to https://help.nextcloud.com

**Comment by qudiqudi** [comment]
I'm also facing this after having dealt with https://github.com/nextcloud/all-in-one/discussions/1787#discussion-4766052 after an update of the AIO container. I don't have this drawio app installed.

´´´
Technical information
=====================
The following list covers which files have failed the integrity check. Please read
the previous linked documentation to learn more about the errors and how to fix
them.

Results
=======
- core
	- INVALID_HASH
		- core/js/mimetypelist.js

Raw output
==========
Array
(
    [core] => Array
        (
            [INVALID_HASH] => Array
                (
                    [core/js/mimetypelist.js] => Array
                        (
                            [expected] => 4b0a99fbab7bc4824176101150bd0ab2b553e128a390e0f872580db4541462b31aad13099fd41533b6695d1b185dabca26a45e9a78227cfb9479eafa439c7bff
                            [current] => a8e818876a7eff2a2137601eafc6c7d96b99cbf27d94ff9bb59eb237bdab92076ecc2828242bb37551c9c8b6772c0718887bd49cee5b49b3be48d3b9ecaa558f
                        )

                )

        )

)

´´´

**Reply by szaimen** [reply]
In your case see https://github.com/nextcloud/all-in-one/discussions/1787#discussioncomment-6121303

**Reply by qudiqudi** [reply]
ah ok, thanks.
Was about to fix it with this https://github.com/nextcloud/server/issues/26022#issuecomment-1575658087

**Reply by szaimen** [reply]
As I said you can ignore the check and it will be fixed automatically with the next Nextcloud update.

**Comment by benoit-smith** [comment]
Hi. Facing the same original issue after updating Nextcloud from 27 to 28.0.1. I just removed the draw.io app, though I guess I have to delete the aforementioned SVG files manually. What about "core/js/mimetypelist.js"? Should I wait for the next update?

**Reply by szaimen** [reply]
> Should I wait for the next update?

yes

**Reply by nordscan** [reply]
[https://help.nextcloud.com/t/http-error-500-after-upgrade-to-28-0-1/177305/4](url)

remove the file and download the latest version then it works,
here are the commands just adjust your installation directory if you have it somewhere else

```
sudo rm /var/www/nextcloud/core/js/mimetypelist.js
sudo wget https://raw.githubusercontent.com/nextcloud/server/master/core/js/mimetypelist.js -
```

**Reply by benoit-smith** [reply]
Thank you all! Upgrading to 28.0.2 solved the issue.

**Reply by ravxen** [reply]
I now have the issue, since I installed draw.io before the 28.0.2 update. After disabling it, the error doesn't change.
Do I have to wait for a new update or can this be fixed manually? And does it effect the usage of nextcloud in any way?

**Reply by W0rmsy** [reply]
Hello, Same issue here

**Comment by pieter-groeneweg** [comment]
It appears not resolved.
on 28.0.4 now

**Comment by moboter** [comment]
After update to Nextcloud 29.0.1
 [EXTRA_FILE] => Array
                (
                    [core/img/filetypes/drawio.svg] => Array
                        (
                            [expected] => 
                            [current] => 92e0974cf869bf8ab969c3442dc2b80d55fde36441d22924db74916a06b407520aa2a9dc39336f9157195ebede697ffac0e639360879255ab91932d406e1897d
                        )

                    [core/img/filetypes/dwb.svg] => Array
                        (
                            [expected] => 
                            [current] => 43731dd5f17a048112ea5109b40b02ec019b3ee2324385a0f448e3bd2264cb13dc160ab018d893f92f8e2f168fd09009b51578c8c6b97a02a1617c67ac087701
                        )
draw.io is installed on the server latest Version (12.3.2-263)
moving the drawio.svg to a new location did not resolve the problem

**Comment by Githopp192** [comment]
@szaimen  - correct - did the nextcloud upgrade from nc 27.1.11 to 28.0.8 ...

after upgrade from 28.0.8 to 29.0.4 - error was gone

	Technical information							
	=====================							
	The following list covers which files have failed the integrity check. Please read							
	the previous linked documentation to learn more about the errors and how to fix							
	them.							
								
	Results							
	=======							
	- core							
	        - INVALID_HASH							
	                - core/js/mimetypelist.js							
	        - EXTRA_FILE							
	                - core/img/filetypes/drawio.svg							
	                - core/img/filetypes/dwb.svg							
								
	Raw output							
	==========							
	Array							
	(							
	    [core] => Array							
	        (							
	            [INVALID_HASH] => Array							
	                (							
	                    [core/js/mimetypelist.js] => Array							
	                        (							
	                            [expected] => 550ab566d30693bfa24ec4b15d9df87731ae8a3be8f79dabf94757e5b8b20eec6e4b678f17af1718297f2872f6b04519eeb024d1dff11947f29da431c7f11201							
	                            [current] => db7fc08a6f59b9db7b245d2ee25f4cf2cbfd70ee0a83a767d587b92baa7663211cfd3bf4a3962ba7d3ec598b3d5153ce027a2d67a0dfbd7f632b2320af47ffb7							
	                        )							
								
	                )							
								
	            [EXTRA_FILE] => Array							
	                (							
	                    [core/img/filetypes/drawio.svg] => Array							
	                        (							
	                            [expected] => 							
	                            [current] => 92e0974cf869bf8ab969c3442dc2b80d55fde36441d22924db74916a06b407520aa2a9dc39336f9157195ebede697ffac0e639360879255ab91932d406e1897d							
	                        )							
								
	                    [core/img/filetypes/dwb.svg] => Array							
	                        (							
	                            [expected] => 							
	                            [current] => 43731dd5f17a048112ea5109b40b02ec019b3ee2324385a0f448e3bd2264cb13dc160ab018d893f92f8e2f168fd09009b51578c8c6b97a02a1617c67ac087701							
	                        )							
								
	                )							
								
	        )							
								
	)
